artifactory/services/utils/properties.go

```diff
@@ -95,10 +96,21 @@ func (props *Properties) AddProperty(key, value string) {
 // If concatValues is true, then the values of each property are concatenated together separated by a comma. For example: key=val1,val2,...
 // Otherwise, each value of the property will be written with its key separately. For example: key=val1;key=val2;...
 func (props *Properties) ToEncodedString(concatValues bool) string {
+	// Sort the keys to create a predictable order
+	keys := make([]string, 0, len(props.properties))
+	for k := range props.properties {
+		keys = append(keys, k)
+	}
+
+	// sort the slice by keys
+	sort.Strings(keys)
+
```

Comment:
To improve readability, I suggest exporting this logic to a local function: `func (props *Properties) getSortedKeys() []string`.